src/java/org/apache/cassandra/utils/ByteBufferUtil.java

```diff
@@ -681,13 +681,13 @@ public static boolean isPrefix(ByteBuffer prefix, ByteBuffer value)
 
     public static boolean canMinimize(ByteBuffer buf)
     {
-        return buf != null && (buf.capacity() > buf.remaining() || !buf.hasArray());
+        return buf != null && (!buf.hasArray() || buf.array().length > buf.remaining());
```

Comment:
Changing from `ByteBuffer#capacity()` to `ByteBuffer#array()` makes it possible to get `UnsupportedOperationException` and `ReadOnlyBufferException` if the byte buffer is not backed by a writable array. If we need the change we should indicate this on the JavaDoc for the method.